tf2onnx/optimizer/transpose_optimizer.py

```diff
@@ -315,7 +323,16 @@ def _create_transpose_pairs_before_node(self, node):
 
         # add Transpose(0, 3, 1, 2) and Transpose(0, 2, 3, 1) before each non_nhwc_trans_consumers
         for input_id, n in non_nhwc_trans_inputs:
-            nchw_node = self._g.make_node("Transpose", [input_id], attr={"perm": [0, 3, 1, 2]})
+            shape = self._g.get_shape(n.output[0])
+            if len(shape) == 4:

```

Comment:
shape is None in my case, current this if cannot handle.
I try to use `shape = n.inputs[0].inputs[0].op.attribute[0].t.dims[:]` to extract the actual shape, this is case by case.
When I try this, it can convert to an onnx model, but running via onnxruntime actually breaks.
The reason is that:
I feel that we cannot simply add `new_shape = [1]*(4-ori_rank) + ori_shape` and then do `Transpose`. This process is not enough. This seems depending on the size of the other input of `Add` .

Reply:
broadcasting op will broadcast inputs to make sure they share same rank, and the broadcast will expand the rank as my code.

Reply:
now the code can accept input with shape info is None